go/cbrotli.go

```diff
@@ -0,0 +1,264 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Distributed under MIT license.
+// See file LICENSE for detail or copy at https://opensource.org/licenses/MIT
+
+// Package cbrotli compresses and decompresses data with C-Brotli library.
+package cbrotli
+
+import (
+	"bytes"
+	"errors"
+	"io"
+
+	"github.com/google/brotli/go/cgo/decoder"
+	"github.com/google/brotli/go/cgo/encoder"
+)
+
+//------------------------------------------------------------------------------
+// Encoder
+//------------------------------------------------------------------------------
+
+// WriterOptions configures Writer.
+type WriterOptions struct {
+	// Quality controls the compression-speed vs compression-density trade-offs.
+	// The higher the quality, the slower the compression. Range is 0 to 11.
+	Quality int
+	// LGWin is the base 2 logarithm of the sliding window size.
+	// Range is 10 to 24. 0 indicates automatic configuration based on Quality.
+	LGWin int
+}
+
+// ErrEncodeFailure is returned when the encode operation fails.
+var ErrEncodeFailure = errors.New("Brotli encode failure")
+
+// ErrWriteAfterClose is returned from Write() if called after Close().
+var ErrWriteAfterClose = errors.New("brotli.Writer: write after close")
+
+// Writer implements io.WriteCloser, an io.Writer decorator that produces
+// Brotli-encoded data.
+type Writer struct {
+	dst     io.Writer
+	encoder encoder.Encoder
+	closed  bool
+}
+
+// NewWriter initializes new Writer instance.
+// Close MUST be called to free resources.
+func NewWriter(dst io.Writer, options WriterOptions) *Writer {
+	return &Writer{
+		dst:     dst,
+		encoder: encoder.New(options.Quality, options.LGWin),
+	}
+}
+
+// Close implements io.Closer. Close() MUST be invoked to free native resources.
+// Also Close() implicitly flushes remaining data to the decorated writer.
+func (z *Writer) Close() error {
+	if z.closed {
+		return nil
+	}
+	defer z.encoder.Close()
+	_, err := z.writeChunk(nil, encoder.Finish)
+	z.closed = true
+	return err
+}
+
+func (z *Writer) writeChunk(p []byte, op encoder.Operation) (int, error) {
+	if z.closed {
+		return 0, ErrWriteAfterClose
+	}
+	var totalBytesConsumed int
+	var err error
+	totalBytesConsumed = 0
+	err = nil
```

Comment:
This is not necessary. 

`var totalBytesConsumed int` and `var err error` implies `totalBytesConsumed == 0 && err == nil`